students/brandon_nguyen/session03/list_lab.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+#Week3 Excercise #2 list_lab
+#Student: Brandon Nguyen - Au2018
+ 
+"""
+"""
+def list_lab():
+    Series1_lst=["Apples","Pears","Oranges","Peaches"]
+ 
+    print(Series1_lst)
+    print()
+ 
+    fruitInput = input("Please enter another fruit and press enter >: ")
+    print()
+    Series1_lst.append(fruitInput)
+    print(Series1_lst)
+    print()
+    #format on remving quote
+    #print ('[%s]' % ', '.join(map(str,Series1_lst)))
+ 
+    indexInput=input("Please enter the number that represents the fruit >: ")
+ 
+    print("\nHere is your Number: \"{}\", and your fruit is: \"{}\"".format(indexInput,Series1_lst[int(indexInput)-1]))
+    print()
+ 
+    #add to list using +
+    Series1_lst=['Grapes'] + Series1_lst
+    print(Series1_lst)
+    print()
+ 
+    Series1_lst.insert(0,"Papaya")
+    print(Series1_lst)
+    print()
+    
+    #print list with p.
+    newList=[]
+    for item in Series1_lst:
+        if item[0]=='P':
+            newList.append(item)
+    print(newList)
+    print()
+#SERIES 2 part
+    Series2_lst=Series1_lst[:]
+    print(Series2_lst)
+    Series2_lst.pop()
+    print(Series2_lst)
+    fruitInputDel = input("Please enter another fruit to delete and press enter >: ")
+  
+    for name in Series2_lst:
+        if name == fruitInputDel:
+            Series2_lst.remove(fruitInputDel)
```

Comment:
You are modifying the list that's being iterated (`Series2_lst`), and the loop's actual behavior might not be what you intended. For example, if `Series2_lst` contains duplicate items that are to be deleted (`name == fruitInputDel`) in consecutive indices, you may find that not all of those duplicate items are deleted with this code.

To avoid that, always iterate over a copy of the list that may be modified. To do that, you can replace the `for` loop header (line 49) to `for name in Series2_lst[:]:`.

Reply:
Thank you for the great way of creating a copy on the fly.